backend/app/services/wiki_service.py

```
@@ -0,0 +1,740 @@
+            wiki_db.refresh(generation)
+            main_db.commit()
+
+            logger.info(f"Cancelled wiki generation {generation_id}")
+
+            return generation
+
+        except HTTPException:
+            wiki_db.rollback()
+            main_db.rollback()
+            raise
+        except Exception as e:
```

Ruff (0.14.7)

Do not catch blind exception: `Exception`

(BLE001)